base/complex.jl

```diff
@@ -524,6 +630,12 @@ function log1p{T}(z::Complex{T})
     end
 end
 
+"""
+    ^(x, y)
+
+Exponentiation operator.
+"""
+Base.:(^)(x, y)
```

Comment:
> Do you know if this message suggests a real problem?

Yes, warnings from Documenter should always point to an actual problem. https://github.com/JuliaLang/julia/pull/19701 will turn these warnings into errors.

The best way I've found to locate where to move docstrings to is to use `methods(func)` and go through the list to see which is the most general definition that applies to the docstring. From the looks of the possible methods I'd say

```
^(x::Number, y::Number) in Base at promotion.jl:194
```

is probably the best choice. So the way to avoid the `No docs found` error is to move the docstring to that definition in `promotion.jl` and then change the signature listed in `doc/src/stdlib/math.md` with the following diff

```diff
diff --git a/doc/src/stdlib/math.md b/doc/src/stdlib/math.md
index e55967911..98d545190 100644
--- a/doc/src/stdlib/math.md
+++ b/doc/src/stdlib/math.md
@@ -9,7 +9,7 @@ Base.:-(::Any, ::Any)
 Base.:*(::Any, ::Any...)
 Base.:(/)
 Base.:\(::Any, ::Any)
-Base.:^(::Any, ::Any)
+Base.:^(::Number, ::Number)
 Base.fma
 Base.muladd
 Base.div
```

Then just repeat the same steps for the other docs that are being moved.

> You can get the docstring for exponentiation at the REPL in any case.

Yes, the REPL help mode isn't as strict about what kind of results it returns; it'll return all possible signatures that match rather than just an exact match. Documenter is just a bit more picky to make it easier to select a specific signature without including all the others that are less specific.